openvdb/points/PointAttribute.h

```diff
@@ -321,7 +321,7 @@ inline void appendAttribute(PointDataTreeT& tree,
     appendAttribute(tree, name, AttributeTypeConversion<ValueType, CodecType>::type(),
         strideOrTotalSize, constantStride, defaultValue, hidden, transient);
 
-    if (!math::isExactlyEqual(uniformValue, Default<ValueType>::value())) {
+    if (!defaultValue || !math::isExactlyEqual(uniformValue, defaultValue->value())) {
```

Comment:
You shouldn't need to run this if `!defaultValue` and `math::isExactlyEqual(uniformValue, Default<ValueType>::value())` (i.e. uniform is 0). Something like:

```
 if (!defaultValue && math::isExactlyEqual(uniformValue, Default<ValueType>::value())) return;
 if (defaultValue  && math::isExactlyEqual(uniformValue, defaultValue->value())) return;
// collapse with uniformValue
```